### ISSUE TITLE
RankEval: Circuitbreaker trips on too many script compilations when using many templated requests

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#21465]
I just tried running more than the usual small-N templated requests as script + params on a blank ES in Kibana and tripped the circuit breaker with this message:

```
"root_cause": [
      {
        "type": "circuit_breaking_exception",
        "reason": "[script] Too many dynamic script compilations within one minute, max: [15/min]; please use on-disk, indexed, or scripts with parameters instead; this limit can be changed by the [script.max_compilations_per_minute] setting",
        "bytes_wanted": 0,
        "bytes_limit": 0
      }
    ]
```

@MaineC you already had an idea what might be wrong here, so filing this as a reminder task.

**Comment by MaineC** [comment]
I'm pretty sure I missed caching the compiled version of the template somewhere, will look into it early next week.

**Comment by MaineC** [comment]
Fixed as part of #21855
